src/components/common/DetectAppVersion.tsx

```diff
@@ -15,53 +16,23 @@ const useStyles = makeStyles((theme: Theme) => ({
   },
 }));
 
-const ONE_MINUTE_INTERVAL_MS = 60 * 1000;
-
 type DetectAppVersionProps = {
   onNewVersion?: () => void;
 };
 
 export default function DetectAppVersion({ onNewVersion }: DetectAppVersionProps): JSX.Element | null {
   const { isMobile } = useDeviceInfo();
   const classes = useStyles();
-  const [lastCheck, setLastCheck] = useState<number>(0);
-  const [needsRefresh, setNeedsRefresh] = useState<boolean>(false);
-  const currentAppVersion = process.env.REACT_APP_TERRAWARE_FE_BUILD_VERSION;
+  const isStale = useAppSelector(selectIsAppVersionStale);
+  const appVersion = useAppSelector(selectAppVersion);
 
   useEffect(() => {
-    let active = true;
```

Comment:
we can see here a great simplification and removal of this complicated business logic that used to be the responsibility of a presentation component. 

Removing the responsibility from this component relieves many issues caused by the ephemeral life cycles of such components.